<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    
    <script>
        const f = ()=>{

        try {
            let a = 0
            // a = b+a
            console.log('run sucsseful')
            // return kiya bhi to bhi finally chalega
            return
        } catch (error) {
            b = a+e
            console.log('error')
        }
        finally{
            //when we do not apply the finally block then and our try  and catch block throug errors then it block this position
            console.log('I am a good boy')
            // it shore to run the fainally block
            // use fo
            //1> close the file
            //2> exit the loop
            //3> clenup
        }
    }

    f()
    </script>
</body>
</html>